src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDsa.cs

```diff
@@ -8,6 +8,20 @@ namespace System.Security.Cryptography
 {
     public abstract class ECDsa : AsymmetricAlgorithm
     {
+        public ECDsa() { }
+
+        public ECDsa(ECNamedCurve curve)
```

Comment:
This allows a similar programming model to new ECDSaCng(keysize) and new ECDsaOpenSSl(keysize). It could actually be on the derived classes independently, but I put it here on the base classs.

It just keeps the oid around temporarily until the key is accessed and lazily created, then it sets the oid value to null. So yes we can get the curve value from just the key.

Example of programming model:
using (cng = new ECDsaCng(ECNamedCurve.brainpoolP160r1)) ...
with old keysize having this:
using (cng = new ECDsaCng(256)) ...